## 12.16 Analysis of Variance 

When fitting statistical models, we may wish to compare how well two models fit the data, to see which is most appropriate. Considering the models we have used in this chapter, we may want to compare (for example):


> Comparison 1: Model 1 (birthweight$\sim$length of pregnancy) vs Model 3 (birthweight$\sim$length of pregnancy+mothers height) 

> Comparison 2: Model 2 (birthweight$\sim$mother's smoking status) vs Model 3 (birthweight$\sim$length of pregnancy+mothers height)

In these examples, Comparison 1 is much simpler than Comparison 2, because the models in Comparison 1 are **nested**. 

Statistical models are said to be **nested** when one model (the simpler model) contains a subset of the covariates in the other one (the complex model) and no other additional variables. In Comparison 2, the models are not nested because the simpler model (Model 2) contains mother's smoking status as a variable, which is not included in Model 3. Nested models can be compared using **Analysis of Variance (ANOVA)** (the comparison of non-nested models is much more complicated and is beyond the scope of this module). 

The main idea of ANOVA is that: if the complex model better describes the data than the simpler model, then we would expect a reasonably large amount of the residual variation that is unexplained by the simpler model to be explained by the complex one. ANOVA provides a statistical framework that can formally test this. 

We will first consider ANOVA in the context of simple linear regression, where the simpler model assumes no association between the outcome and the independent variable (the **null** model). We will then consider ANOVA in the context of multivariable linear regression and we end by learning how ANOVA can be used to test for differences between groups in a categorical variable. 

### 12.16.1 ANOVA for simple linear regression 

The total variation in $Y$ (otherwise known as the **sum of squares of the $Y$'s**), is denoted by $SS_{TOT}$ and is equal to:

$$ 
SS_{TOT} = \sum_{i=1}^n (y_i-\bar{y})^2
$$

This can be partitioned into two components: the predictable variation in $Y$ ($SS_{REG}$) and the unpredictable variation in $Y$ ($SS_{RES}$). A key result for ANOVA is:

$$
SS_{TOT} = SS_{REG}+SS_{RES} \\
\rightarrow \sum_{i=1}^n (Y_i-\bar{Y})^2 = \sum_{i=1}^n (\hat{Y_i}-\bar{Y})^2 + \sum_{i=1}^n(Y_i-\hat{Y_i})^2
$$

In the above equations, $SS_{TOT}$ (i.e the TOTal sum of squares), represents all of the variation in $Y$ about its mean value. $SS_{REG}$ (i.e. the REGression sum of squares) represents the variation of the predicted values $\hat{Y}$ about the mean. $SS_{RES}$ (i.e. the RESidual sum of squares) represents the variation of the observed values about their predicted values. 

Using the sums of squares defined above, we can calculate the proportion of variance explained by statistical model, known as the **coefficient of determination**. 

### 12.16.2 The coefficient of determination

The proportion of variation which is explained by a statistical model is denoted by $R^2$ and is given by: 

$$
R^2 = \frac{SS_{REG}}{SS_{TOT}}.
$$

*Example.* The coefficient of determination is given in the ```summary()``` output for a linear regression in R. In Model 1, $R^2=0.1661$ (see output below). This means that Model 1 explains 16.6\% of the total variation in $Y$.

In [2]:
#The coefficient of determination
data<- read.csv('https://www.inferentialthinking.com/data/baby.csv')
model1<-lm(Birth.Weight~Gestational.Days, data=data)
summary(model1)


Call:
lm(formula = Birth.Weight ~ Gestational.Days, data = data)

Residuals:
    Min      1Q  Median      3Q     Max 
-49.348 -11.065   0.218  10.101  57.704 

Coefficients:
                  Estimate Std. Error t value Pr(>|t|)    
(Intercept)      -10.75414    8.53693   -1.26    0.208    
Gestational.Days   0.46656    0.03054   15.28   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 16.74 on 1172 degrees of freedom
Multiple R-squared:  0.1661,	Adjusted R-squared:  0.1654 
F-statistic: 233.4 on 1 and 1172 DF,  p-value: < 2.2e-16


While $R^2$ is sometimes used an overall measure of goodness-of-fit (or predictive performance), it isn't used to formally compare models. This is because $R^2$ will never decrease when new covariates are added to a model (provided that the number and identity of observations remains the same). Therefore, using $R^2$ for model comparisons, we would always conclude that the more complex model is at least as good a fit as the simpler model, even if this is not true. We can calculate an adjusted $R^2$ (the adjusted $R^2$ for Model 1 is given in the output above) that accounts for this issue. Alternatively, we can use analysis of variance (ANOVA). To describe ANOVA, we need to first define the remaining statistics that are commonly summarised in what is known as the ANOVA table. 

### 12.16.3 The ANOVA table. 

Each of the sum of squares defined above ($SS_{TOT}$, $SS_{REG}$ and $SS_{RES}$) have an associated **degrees of freedom (d.f.)**. The d.f. for the total sum of squares is $(n-1)$, since the variance of $Y$ is $\sum_{i=1}^n (Y_i-\bar{Y})^2/(n-1)$. The d.f. for the regression sum of squares in the number of covariates in the regression model (when a simple linear regression model is used this is equal to 1). The residual d.f. is found by subtracting the regression d.f. from the total d.f. The sums of squares also have associated mean squares, which are obtained by dividing the sum of squares by its associated degrees of freedom (note that the residual mean square is then equal to $\hat{\sigma}^2$). These statistics are summarised in an ANOVA table for simple linear regression (Table 2). 


Source      | d.f.  | SS         | Mean Square                        | 
------------|-------|------------|------------------------------------|
Regression  | 1     | $SS_{REG}$ | $MS_{REG}=\frac{SS_{REG}}{1}$      |
Residual    | n-2   | $SS_{RES}$ | $MS_{RES}=\frac{SS_{RES}}{n-2}$  | 
Total       | n-1   | $SS_{TOT}$ | $MS_{TOT}=\frac{SS_{TOT}}{n-1}$    | 

Table 2: The ANOVA Table for simple linear regression 

The values in this table can be used to conduct formal hypothesis tests. 

### 12.16.4 Hypothesis testing using ANOVA

ANOVA is used to test the null hypothesis that the simpler of the two nested models better fits the data. In simple linear regression, the simpler model is the null model, in which case:

+ $H_0:$ The null model is a better fit
+ $H_1:$ The simple linear regression model is a better fit

To test the null hypothesis defined above, we use an $F$ statistic, defined as:

$$
F = \frac{MS_{REG}}{MS_{RES}}
$$

This ratio measures how much more variation in $Y$ is explained by the model than would be expected by chance. If the model does not fit the data well, then we would expect this ratio to be equal to 1. The larger the value of $F$, the stronger the evidence that the complex model is a better fit. To obtain a $p$-value for a formal hypothesis test, $F$ can be compared to the $F_{1,(n-2)}$ distribution (where 1 and (n-2) are the relevant degrees of freedom for the mean squares).

### 12.16.5 Similarity between F tests and t-tests in simple linear regression 

Since the null model is defined as: $Y=\beta_0+\epsilon$ and the simple linear regression model is defined as: $Y=\beta_0+\beta_1X+\epsilon$. The hypotheses above can be rewritten as:

+ $H_0: \beta_1 = 0$ 
+ $H_1: \beta_1 \neq 0$ 

In other words, the $F$-test for a simple linear regression model is the same as the $t$-test of the null hypothesis that the slope parameter is equal to 0. Indeed, the two tests are equivalent with $F=t^2$. Consequently, it is not particularly common to use $F$-tests in the simple linear regression model, they are more useful for assessing more complex models with multiple covariates. 

### 12.16.6 Example using Model 1

The $F$ test statistic and associated $p$-value are given in the summary output for linear models (shown below):

In [15]:
# F-test
summary(model1)


Call:
lm(formula = Birth.Weight ~ Gestational.Days, data = data)

Residuals:
    Min      1Q  Median      3Q     Max 
-49.348 -11.065   0.218  10.101  57.704 

Coefficients:
                  Estimate Std. Error t value Pr(>|t|)    
(Intercept)      -10.75414    8.53693   -1.26    0.208    
Gestational.Days   0.46656    0.03054   15.28   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 16.74 on 1172 degrees of freedom
Multiple R-squared:  0.1661,	Adjusted R-squared:  0.1654 
F-statistic: 233.4 on 1 and 1172 DF,  p-value: < 2.2e-16


Alternatively, we can use ```anova()``` which outputs the ANOVA table as well as the relevant $F$ statistic and $p$-value:

In [16]:
# F-test using anova()
anova(model1)

Df   Sum Sq    Mean Sq    F value  Pr(>F)      
Gestational.Days    1  65449.51 65449.5131 233.4293 3.395226e-48
Residuals        1172 328608.34   280.3825       NA           NA

As can be seen from both outputs above, the $F$-statistic is equal to 233.4 with $p$-value $<2.2\times10^{-16}$. With such a small $p$-value there is strong evidence against the null and therefore we conclude that the model which includes gesational days is a better fit (i.e. we conclude that $\beta_1 \neq 0$). 

Notice that in the ```summmary()``` output, the $p$-values for the F-test and the t-test for gestational days are identical, and that $F=233.4=15.24^2=t^2$. 
